### PR TITLE
FEATURE: Recipe For The Day Suggestion section

### LINKED ISSUE #328: FEATURE: Recipe For The Day Suggestion section
### Is there an existing issue raised for this?

- [X] I have searched the existing issues

### Describe the feature

Random Recipe: A feature that suggests a random recipe for users looking for inspiration.

### Component

I would like to add this feature on Homepage.

### It should be implemented because

Enhanced User Engagement: By showcasing a new, random recipe daily, users are encouraged to explore recipes they might not otherwise discover, increasing the time they spend on the platform.

Fresh and Dynamic Content: The homepage remains dynamic and interesting with new content appearing daily, making users more likely to return frequently to see what's new.

Recipe Discovery: It promotes lesser-known or newly contributed recipes, giving them more visibility and helping users broaden their culinary horizons.

User Inspiration: For users unsure of what to cook, the "Recipe of the Day" serves as a quick source of inspiration, solving decision fatigue.

### Would you like to work on this issue?

Yes

### Terms & Conditions

- [X] I agree to follow the Code of Conduct
- [X] I'm a GSSOC'24 Ext contributor

### PR DESCRIPTION
**Pull request by sanchitc05**
Fixes #328

## What does this PR do?

This PR implements a "Recipe of the Day" feature for TastyTrails. It includes:

1. Backend changes:
   - New fields in the Recipe model: `featured` and `lastFeatured`
   - New controller function `getRecipeOfTheDay`
   - New route `/recipe-of-the-day`

2. Frontend changes:
   - New `RecipeOfTheDay` component
   - Integration of the new component in the Landing page

The feature selects a new recipe each day, avoiding repetition within 7 days when possible. It provides users with daily culinary inspiration directly on the landing page.

Fixes #328 (Implement Recipe of the Day feature)

## How should this be tested?

1. Backend Testing:
   - Start the backend server
   - Use a tool like Postman to send a GET request to `/api/recipe-of-the-day`
   - Verify that a recipe is returned with `featured: true`
   - Run this test on multiple days to ensure a different recipe is featured

2. Frontend Testing:
   - Start the frontend application
   - Navigate to the landing page
   - Scroll down to find the "Recipe of the Day" section
   - Verify that a recipe is displayed with its image, name, and description
   - Click the "View Recipe" button and confirm it navigates to the full recipe page
   - Check the responsiveness on different screen sizes

3. Long-term Testing:
   - Monitor for 7+ days to ensure recipes are not repeated within a week unless necessary

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read Contributing guide
- [x] Self-reviewed my code
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] The changes don't cause any responsiveness issues

### CODE REVIEW

**Comment by AlfiyaSiddique** [comment]
@sanchitc05 can you please add the video of the output here

**Comment by AlfiyaSiddique** [comment]
Also can you test with the case of date change so that every day different recipe is shown?

**Comment by sanchitc05** [comment]
Sure thing @AlfiyaSiddique I'll share soon

**Comment by AlfiyaSiddique** [comment]
@sanchitc05 any updates?

**Comment by sanchitc05** [comment]
Yeah @AlfiyaSiddique I'll just do it by today night or tomorrow morning, I had been busy in the past days

**Comment by AlfiyaSiddique** [comment]
Sure @sanchitc05 take your time.

**Comment by MastanSayyad** [comment]
@AlfiyaSiddique Review this and merge if possible

**Comment by AlfiyaSiddique** [comment]
@sanchitc05  ??

https://github.com/AlfiyaSiddique/TastyTrails/pull/420#issuecomment-2460714879
https://github.com/AlfiyaSiddique/TastyTrails/pull/420#issuecomment-2460717864

**Comment by AlfiyaSiddique** [comment]
>@AlfiyaSiddique Review this and merge if possible

@MastanSayyad the change is not completed yet I think; no update from the contributor's side.
Can't merge without testing

**Comment by MastanSayyad** [comment]
@sanchitc05 do the needful, As the PA is active make sure you contact her before you contact us.

**Comment by sanchitc05** [comment]
@AlfiyaSiddique TBH my pc got a blue screen error and it is out for repair, I took my friend's laptop but could not run it properly as I was getting various dependencies errors. I would just request you to see the code files, I assure you the logic is right and I have integrated it with react so that the output can be seen under the best dishes section. If you want to check the output you could just merge it and then check it directly deployed on netlify, in case you need any modifications you can always revert my pr and I can make changes to it.

**Comment by AlfiyaSiddique** [comment]
Okay @sanchitc05 works

File: backend/Controllers/RecipeController.js [hunk 1]
```diff
@@ -647,3 +647,58 @@ const RecipeController = {
...
+
+  } catch (error) {
+    console.error('Error in getRecipeOfTheDay:', error);
+    res.status(500).json({
+      success: false,
+      message: "Error fetching Recipe of the Day",
+      error: error.message
+    });
+  }
+};
```
Comment (AlfiyaSiddique) [inline backend/Controllers/RecipeController.js:704]: @sanchitc05  you have written the function but it is not exported, how will it work?

File: backend/routes/web.js [hunk 2]
```diff
@@ -52,3 +52,5 @@ router.delete(
 
 router.patch("/recipe/share/:recipeId", authenticateToken, RecipeController.updateShareCount);
 export default router;
+
+router.get('/recipe-of-the-day', getRecipeOfTheDay);
```
Comment (AlfiyaSiddique) [inline backend/routes/web.js:56]: @sanchitc05 the way the function is added to the route is wrong please refer other routes and always put everything before the export in all changed files

Comment (sanchitc05) [reply backend/routes/web.js:56]: Okay @AlfiyaSiddique

File: backend/Controllers/RecipeController.js (hunk 1 above)
Comment (sanchitc05) [reply backend/Controllers/RecipeController.js:704]: Let me make the necessary changes

**Comment by AlfiyaSiddique** [comment]
@sanchitc05 There are logical issues in the backend code. I have added a comment please fix it.

**Comment by sanchitc05** [comment]
@AlfiyaSiddique On it!

**Comment by sanchitc05** [comment]
@AlfiyaSiddique  I'm closing this pr and i will make a new pr doing the necessary modifications that you suggested.

**Pull request closed without merging**